### PR TITLE
Fix random strategy behaviour

### LINKED ISSUE #2244: Random strategy moves services constantly
Regression in 1.2.x

### PR DESCRIPTION
**Pull request by jakolehm**
Fixes #2244 

Includes also cleanup of scheduler internals.

### CODE REVIEW

File: server/app/services/scheduler/strategy/random.rb [hunk 1]
```diff
@@ -1,34 +1,58 @@
...
+      # @return [Scheduler::Node]
+      def find_stateless_node(grid_service, instance_number, nodes)
+        if deployment?
+          nodes.sample
+        else 
           prev_instance = grid_service.grid_service_instances.find_by(
             grid_service: grid_service, instance_number: instance_number
           )
           if prev_instance
-            prev_instance.host_node
```
Comment (SpComb) [inline server/app/services/scheduler/strategy/random.rb]: This also fixes #2254

**Comment by jakolehm** [comment]
Going to open separate pr for 1.2.2.

**Comment by SpComb** [comment]
Needs merge after #2256 

I think this PR is still useful for the `GridServiceDeployWorker` and `GridServiceDeployer` -> `GridServiceScheduler` refactoring on its own, but I don't really understand what we need the automatic/manual deployment modes and scheduler/deployment scheduling modes for, since we fixed the only usage of those in #2256

**Comment by jnummelin** [comment]
Do we really need this for 1.3 as the behaviour if already fixed in 1.2.2?

**Comment by SpComb** [comment]
I don't think this belongs under the 1.3.X milestone... tentatively bumping to 1.4 for the refactoring, but this PR seems out of date, so perhaps just close it...?

**Comment by SpComb** [comment]
Closing as obsolete. Wouldn't mind if we picked up the refactoring bits as a new PR.

**Pull request closed without merging**